Unpin pylint version

> Can an issue be opened to fix the linting error and remove the pin, with an example of what needs to be fixed.
>
> Issue linked here so it's not lost, and is connected to the PR
>
> _Originally posted by @cidrblock in https://github.com/ansible/ansible-navigator/issues/538#issuecomment-902095978_

Pylint is temporarily capped in https://github.com/ansible/ansible-navigator/pull/538 and to drop the limitation we have to fix `R1732 (consider-using-with)`.

The exact failures upgrade produces related to the new rule `consider-using-with` that appeared in v2.9 are:
```console
************* Module ansible_navigator.actions.open_file
ansible_navigator/actions/open_file.py:91:27: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
ansible_navigator/actions/open_file.py:95:27: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
ansible_navigator/actions/open_file.py:98:27: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
ansible_navigator/actions/open_file.py:102:27: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
```

--- a/test-requirements.txt
+++ b/test-requirements.txt
@@ -3,6 +3,6 @@ black
 libtmux
 lxml
 pre-commit
-pylint < 2.9  # v2.9.0 introduced a check for `with` which is not yet fixed
+pylint
 pytest-cov
 pytest-xdist

--- a/tox.ini
+++ b/tox.ini
@@ -85,7 +85,7 @@ commands =
     'print(f"\n\{sep\}\nTo install pre-commit hooks into the Git repo, run:\n\n\{cmd_str\}\n\n\{sep\}\n")'
 deps =
   pre-commit
-  pylint ~= 2.8.0
+  pylint
   pylint-pytest < 1.1.0
 isolated_build = true
 skip_install = true

